docs/reference/mapping/types/range.asciidoc

```diff
@@ -229,3 +235,208 @@ The following parameters are accepted by range types:
+be changed or removed in a future release. Elastic will work to fix
+any issues, but features in technical preview are not subject to the support SLA
+of official GA features.
+
+`range` fields support <<synthetic-source,synthetic `_source`>> in their default
+configuration. Synthetic `_source` cannot be used together with
+<<copy-to,`copy_to`>> or with <<doc-values,`doc_values`>> disabled.
+
+Synthetic source always sorts values and removes duplicates for all `range` fields except `ip_range` . Ranges are sorted by their lower bound and then by upper bound. For example:
+[source,console,id=synthetic-source-range-sorting-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": { "mode": "synthetic" },
+    "properties": {
+      "my_range": { "type": "long_range" }
+    }
+  }
+}
+PUT idx/_doc/1
+{
+  "my_range": [
+    {
+        "gte": 200,
+        "lte": 300
+    },
+    {
+        "gte": 1,
+        "lte": 100
+    },
+    {
+        "gte": 200,
+        "lte": 300
+    },
+    {
+        "gte": 200,
+        "lte": 500
+    }
+  ]
+}
+----
+// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]
+
+Will become:
+
+[source,console-result]
+----
+{
+  "my_range": [
+    {
+        "gte": 1,
+        "lte": 100
+    },
+    {
+        "gte": 200,
+        "lte": 300
+    },
+    {
+        "gte": 200,
+        "lte": 500
+    }
+  ]
+}
+----
+// TEST[s/^/{"_source":/ s/\n$/}/]
+
+Values of `ip_range` fields are not sorted but original order is not preserved. Duplicate ranges are removed. If `ip_range` field value is provided as a CIDR, it will be represented as a range of IP addresses in synthetic source.
+
+For example:
+[source,console,id=synthetic-source-range-ip-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": { "mode": "synthetic" },
+    "properties": {
+      "my_range": { "type": "ip_range" }
+    }
+  }
+}
+PUT idx/_doc/1
+{
+  "my_range": [
+    {
+      "gte": "192.168.0.0",
+      "lte": "192.168.0.20"
+    },
+    "10.0.0.0/24",
+    {
+      "gte": "10.0.0.0",
+      "lte": "10.0.0.255"
+    }
+  ]
+}
+----
+// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]
+
+Will become:
+
+[source,console-result]
+----
+{
+  "my_range": [
+    {
+      "gte": "10.0.0.0",
+      "lte": "10.0.0.255"
+    },
+    {
+      "gte": "192.168.0.0",
+      "lte": "192.168.0.20"
+    }
+  ]
+}
+----
+// TEST[s/^/{"_source":/ s/\n$/}/]
+
+Ranges are stored as inclusive on both sides so in synthetic source they are represented in this format as well. For example:
+[source,console,id=synthetic-source-range-normalization-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": { "mode": "synthetic" },
+    "properties": {
+      "my_range": { "type": "long_range" }
+    }
+  }
+}
+PUT idx/_doc/1
+{
+  "my_range": {
+    "gt": 200,
+    "lt": 300
+  }
+}
+----
+// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]
+
+Will become:
+
+[source,console-result]
+----
+{
+  "my_range": {
+    "gte": 201,
+    "lte": 299
+  }
+}
+----
```

Comment:
Maybe we should add this to the list of limitations for `synthetic source`? docs/reference/mapping/fields/synthetic-source.asciidoc

Reply:
This is specific to ranges and i followed existing docs that have specific examples on type page (for example `ip`).

Reply:
I am open to moving it though.